.github/dependabot.yml

```diff
@@ -94,8 +95,6 @@ updates:
     include: scope
   ignore:
   - dependency-name: '*golang*'
-    update-types:
-    - version-update:semver-major
```

Comment:
Is the idea here that we want dependabot to tell us when a major update lands downstream, so we can either just scoop it in or manually fix up everything assuming interfaces have changed, etc?

Reply:
I think we want to ignore all updates from dependabot for this since we will passively consume patch updates given the version in the Dockerfile is a floating MAJOR.MINOR tag, minor updates we stay in sync with CAPI, and major updates are not on the table for the foreseeable future and obviously would require more intensive planning.

A new minor release could then maybe slip through the cracks for a while but historically that hasn't been an issue. Even probably even then it probably wouldn't cause any real issues until the version of Go in the base image goes out of support months down the road and gets CVEs. Probably some dependency we pull in or a user importing any packages will require an updated Go and force us to update before then.

Reply:
Maybe I'm not understanding this change, as to me this change goes from:

- do not permit major version updates

to:

- permit major version updates

Reply:
This change goes from "ignore major version updates" to "ignore all updates."

Reply:
https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#specifying-dependencies-and-versions-to-ignore

Reply:
I'm actually not 100% sure of the semantics here, but I know that the go.mod dependencies configured here don't include any `update-types` and here we do and we still see PRs like #4559, so this is an attempt to stop those PRs.

Reply:
That is not intuitive, thanks for doing that dependabot behavior discovery work on my behalf! ❤️

btw it looks like "docker.*golang*" at the root dir will be responsible for `/Dockerfile`, and I guess we don't worry about patch versions because docker is doing that for us already so long as we use something like `FROM golang:1.20`.